RecoVertex/BeamSpotProducer/plugins/OnlineBeamSpotESProducer.cc

```diff
@@ -129,13 +133,22 @@ const BeamSpotOnlineObjects* OnlineBeamSpotESProducer::checkSingleBS(const BeamS
   auto limitTime = std::chrono::microseconds((std::chrono::hours)timeThreshold_).count();
 
   // Check that the BS is within the timeThreshold, converges and passes the sigmaZthreshold
-  if (diffBStime1 < limitTime && bs1->sigmaZ() > sigmaZThreshold_ && bs1->beamType() == 2) {
+  if (diffBStime1 < limitTime && isGoodBS(bs1)) {
     return bs1;
   } else {
     return nullptr;
   }
 }
 
+// This method is used to check the quality of the beamspot fit
+const bool OnlineBeamSpotESProducer::isGoodBS(const BeamSpotOnlineObjects* bs1) {
+  if (bs1->sigmaZ() > sigmaZThreshold_ && bs1->beamType() == 2 && bs1->beamWidthX() > sigmaXYThreshold_ * 1E-4 &&
+      bs1->beamWidthY() > sigmaXYThreshold_ * 1E-4)
```

Comment:
The `* 1E-4` conversion could be done only a single time in the constructor, instead of once/twice per beamspot ?